### PR TITLE
feat: add support for --secret

### PR DESCRIPTION
**Pull request by colesnodgrass**
- add support for creating secrets via the `--secret` flag
   ```
   abctl local install --secret=/path/to/secrets.yml
   ```
   ```yml
   # secrets.yml
   apiVersion: v1
   kind: Secret
   metadata:
     name: secret-name
   stringData:
     key: value
   ```
    - multiple `--secret` flags can be provided
       ```
       abctl local install --secret=/path/to/secret-a.yml --secret=/path/to/secret-b.yml
       ```   
- update error messages for consistency

### CODE REVIEW

**Review by bgroff — APPROVED** [review]

**Comment by killthekitten** [comment]
@bgroff @colesnodgrass does this support providing multiple secrets files? In our case we only need one (the external DB) but in some of the docs I saw other examples of secrets. Basic Auth password or S3 api key come to mind

**Comment by colesnodgrass** [comment]
> @bgroff @colesnodgrass does this support providing multiple secrets files? In our case we only need one (the external DB) but in some of the docs I saw other examples of secrets. Basic Auth password or S3 api key come to mind

@killthekitten 
Currently no, but I'm planning to add this in one of two ways:
1. enable the `--secrets` flag to support a file with multiple secrets (separated by `---`)
2. rename this to `--secret` and allow you to pass it multiple times, each pointing to a different file

**Comment by colesnodgrass** [comment]
Updated to use `--secret`, which now can be provided multiple times for multiple secrets.

**Review by bgroff — APPROVED** [review]

File: internal/cmd/local/local/cmd.go [hunk 1]
```diff
@@ -545,26 +570,44 @@ func (c *Command) handleBasicAuthSecret(ctx context.Context, user, pass string)
 		return fmt.Errorf("unable to hash basic auth password: %w", err)
 	}
 
-	data := map[string][]byte{"auth": []byte(fmt.Sprintf("%s:%s", user, hashedPass))}
-	if err := c.k8s.SecretCreateOrUpdate(ctx, corev1.SecretTypeOpaque, airbyteNamespace, "basic-auth", data); err != nil {
-		pterm.Error.Println("Could not create Basic-Auth secret")
+	secret := corev1.Secret{
```
Comment (perangel) [inline internal/cmd/local/local/cmd.go:573]: For a later refactor: we can add a util for instantiating a secret object as we continue to add more cases.

File: internal/cmd/local/local_install.go [hunk 2]
```diff
@@ -149,25 +151,12 @@ func NewCmdInstall(provider k8s.Provider) *cobra.Command {
...
-				if env := os.Getenv(envDockerUser); env != "" {
-					opts.DockerUser = env
-				}
-				if env := os.Getenv(envDockerPass); env != "" {
-					opts.DockerPass = env
-				}
-				if env := os.Getenv(envDockerEmail); env != "" {
-					opts.DockerEmail = env
-				}
+				envOverride(&opts.BasicAuthUser, envBasicAuthUser)
```
Comment (perangel) [inline internal/cmd/local/local_install.go:154]: 👍🏼 nice, much cleaner

**Review by perangel — COMMENTED** [review]
Looks good 👍🏼

**Review by perangel — APPROVED** [review]

**Pull request merged**